src/main/java/one/nio/lz4/LZ4.java

```diff
@@ -536,11 +544,11 @@ private static int decompress(final Object src, final long srcOffset,
             int token = unsafe.getByte(src, ip++) & 0xff;
             int length = token >>> ML_BITS;
             if (length == RUN_MASK) {
-                int s;
-                do {
+                int s = 255;
+                while (ip < srcEnd - RUN_MASK && s == 255) {
```

Comment:
is `decompress_Incomplete` cover this change?

Reply:
If yes, please be aware about this check, most likely first branch in native code is executed:

        if (NativeLibrary.IS_SUPPORTED) {
            result = decompress0(src, srcOffset, dst, dstOffset, length, dst.length - dstOffset);
        } else {
            result = decompress(src, byteArrayOffset + srcOffset, dst, byteArrayOffset + dstOffset, length, dst.length - dstOffset);
        }

Reply:
@Marcono1234 I think we can add environment flag to choose testing branch or simply  make 
```
private static int decompress
```
package local and directly call it in test